public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FormEvent } from 'react';
+import { useDispatch } from 'react-redux';
+import { Icon, Input, Label, RadioButtonGroup, useStyles } from '@grafana/ui';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { css } from 'emotion';
+
+import DataSourceSelect from '../DataSourceSelect';
+import { rulesFiltersSlice } from '../../state/reducers';
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+
+const RulesFilter = () => {
+  const dispatch = useDispatch();
+  const { rulesFilters } = useUnifiedAlertingSelector((state) => state.filters);
```

Comment:
I think there's no reason to put filter state into redux. Just use query params as the single source of truth.
Less boilerpalte and state sync issues are avoided

Reply:
Agreed